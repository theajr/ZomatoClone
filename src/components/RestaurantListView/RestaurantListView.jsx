import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import Restaurant from '../Restaurant/Restaurant';
import Loader from '../Loader';

const RestaurantListView = ({ loading, restaurants, error }) => {
  if (loading) return <Loader />;
  if (!restaurants.length)
    return (
      <Typography
        component="h5"
        variant="h5"
        style={{ paddingTop: 50, textAlign: 'center' }}
      >
        No Results matching your search/filter criteria
      </Typography>
    );
  const list = restaurants.map(restaurant => (
    <Restaurant restaurant={restaurant} key={restaurant.id} />
  ));
  return <div>{list}</div>;
};

RestaurantListView.propTypes = {};

export default RestaurantListView;
