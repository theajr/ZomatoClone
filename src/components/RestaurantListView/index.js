import { connect } from 'react-redux';
import _ from 'lodash';
import RestaurantListView from './RestaurantListView';

const filterRestaurants = (res, selectedCuisines, ratingRange) => {
  const rating = parseInt(res.user_rating.aggregate_rating, 10);
  return res.cuisines.some(
    c =>
      (selectedCuisines.length === 0 || selectedCuisines.includes(c)) &&
      rating >= ratingRange[0] &&
      rating <= ratingRange[1]
  );
};
// _.orderBy(users, ['user', 'age'], ['asc', 'desc']);
const mapStateToProps = ({ restaurant, sortSearchFilterOptions }) => {
  const filtered = restaurant.restaurants.filter(res =>
    filterRestaurants(
      res,
      sortSearchFilterOptions.search.cuisines,
      sortSearchFilterOptions.filter.ratingRange
    )
  );
  const sorted = _.orderBy(
    filtered,
    ['user_rating.aggregate_rating'],
    [sortSearchFilterOptions.sort.rating.toLowerCase()]
  );

  return {
    ...restaurant,
    restaurants: sorted,
  };
};

const mapDispatchToProps = dispatch => ({});

export default connect(mapStateToProps, mapDispatchToProps)(RestaurantListView);
