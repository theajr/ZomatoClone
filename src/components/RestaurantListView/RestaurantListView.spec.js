import React from 'react';
import { shallow } from 'enzyme';
import RestaurantListView from './RestaurantListView';

describe('<RestaurantListView />', () => {
  test('renders loader', () => {
    const wrapper = shallow(<RestaurantListView loading />);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders list', () => {
    const wrapper = shallow(<RestaurantListView restaurants={[{}]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
