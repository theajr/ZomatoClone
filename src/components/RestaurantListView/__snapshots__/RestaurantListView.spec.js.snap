// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantListView /> renders 1`] = `ShallowWrapper {}`;

exports[`<RestaurantListView /> renders list 1`] = `ShallowWrapper {}`;

exports[`<RestaurantListView /> renders loader 1`] = `ShallowWrapper {}`;
